lib/hooks/useSlots.ts

```diff
@@ -0,0 +1,138 @@
+import { useState, useEffect } from "react";
+import getSlots from "@lib/slots";
+import { User, SchedulingType } from "@prisma/client";
+import dayjs, { Dayjs } from "dayjs";
+import isBetween from "dayjs/plugin/isBetween";
+import utc from "dayjs/plugin/utc";
+dayjs.extend(isBetween);
+dayjs.extend(utc);
+
+type Slot = {
+  time: Dayjs;
+  users?: string[];
+};
+
+type UseSlotsProps = {
+  eventLength: number;
+  minimumBookingNotice?: number;
+  date: Dayjs;
```

Comment:
I've had a think about this, and because this is like it used to be I'm hesitant to change it now; Soon we'll embark on the virtual calendar journey and I don't think useSlots will be left unscaled in that optimisation. Let's do this then.